File: Algorithm-CPP/13_가장많이사용된자릿수.cpp
#include <iostream>
using namespace std;

int main() {
    int n, nums[] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0};

    cin >> n;

    while(n != 0) {
        int index = n % 10;
        n /= 10;
        nums[index]++;
    }

    int max = nums[0];
    int result;

    for(int i = 1; i<10; i++) {
        if(nums[i] > max) {
            max = nums[i];
            result = i;
        } else if(nums[i] == max) {
            result = i;
        }
    }

    cout << result << endl;

    return 0;
}

File: Algorithm-Swift/Algorithm-Swift/Inflearn/13_가장많이사용된자릿수.swift
//
//  13_가장많이사용된자릿수.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/07/29.
//

import Foundation

func solution13() {
    let str = readLine()!
    var counts = Array(repeating: 0, count: 10)
    var max = 0, result = 0

    str.forEach { s in
        let index = Int("\(s)")!
        counts[index] += 1
    }

    counts.enumerated().forEach { index, num in
        if num > max {
            max = num
            result = index
        } else if num == max {
            result = index
        }
    }

    print(result)
}
